normalize_label_bench_test.go

```diff
@@ -0,0 +1,23 @@
+package prometheus
+
+import "testing"
+
+var labelBenchmarkInputs = []string{
+	"",
+	"label:with:colons",
+	"LabelWithCapitalLetters",
+	"label!with&special$chars)",
+	"label_with_foreign_characters_字符",
+	"label.with.dots",
+	"123label",
+	"_label_starting_with_underscore",
+	"__label_starting_with_2underscores",
+}
+
+func BenchmarkNormalizeLabel(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		for _, input := range labelBenchmarkInputs {
+			NormalizeLabel(input)
+		}
+	}
```

Comment:
You should have a sub-benchmark per case:
```suggestion
    for _, input := range labelBenchmarkInputs {
    	b.Run(input, func(b *testing.B) {
    	    	NormalizeLabel(input)
    	})
    }
```